build/images/codegen/Dockerfile

```diff
@@ -54,3 +54,6 @@ RUN go install k8s.io/code-generator/cmd/client-gen@kubernetes-$K8S_VERSION && \
 COPY --from=protoc /tmp/protoc/bin /usr/local/bin
 COPY --from=protoc /tmp/protoc/include /usr/local/include
 
+# workaround for safe directory issue on github actions
+# ref: https://github.com/actions/runner-images/issues/6775
+RUN git config --global --add safe.directory /go/src/antrea.io/antrea
```

Comment:
I'm fine with the solution overall, but I think we can make it a bit nice:

1) Avoid hardcoding the `/go/src/antrea.io/antrea` in the Docker file. Instead of using a `RUN` instruction in the Dockefile, let's update the `update-codegen.sh` script so that it looks like this:

```bash
set -o errexit
set -o pipefail

ANTREA_ROOT="$( cd "$( dirname "${BASH_SOURCE[0]}" )/../" && pwd )"
IMAGE_NAME="antrea/codegen:kubernetes-1.26.4"

function docker_run() {
  docker pull ${IMAGE_NAME}
  set -x
  ANTREA_PATH="/go/src/antrea.io/antrea"
  docker run --rm \
		-e GOPROXY=${GOPROXY} \
		-w ${ANTREA_PATH} \
		-v ${ANTREA_ROOT}:${ANTREA_PATH} \
		"${IMAGE_NAME}" bash -c "git config --global --add safe.directory ${ANTREA_PATH} && $@"
}

docker_run hack/update-codegen-dockerized.sh "$@"
```
2. Add a better comment in `update-codegen.sh` to explain why this is needed. Pointing to that issue is a bit misleading as this is orthogonal to Github actions. Maybe something like this:

> Recent versions of Git will not access .git directories which are owned by another user (as a security measure), unless the directories are explicitly added to a "safe" list in the Git config. When we run the Docker container, the Antrea source directory may be owned (depends on the Docker platform) by a user which is different from the container user (as the source directory is mounted from the host). If this is the case, the Git program inside the container will refuse to run. This is why we explicitly add the Antrea source directory to the list of "safe" directories. We are still looking into the possibility of running the Docker container as the "current host user".

Reply:
Thanks for the suggestions. Done.